File: python/applications/stickstick/resources/data/note.py
import sys
sys.path.append('%s/../libraries/' % prudence.source.basePath)

from sqlalchemy.orm.exc import NoResultFound
import minjson as json

prudence.execute('../libraries/stickstick/data/')

def get_id(conversation):
    try:
        return int(conversation.resource.request.attributes.get('id'))
    except ValueError:
        return None
    except TypeError:
        return None
    #form = conversation.resource.request.resourceRef.queryAsForm
    #return int(form.getFirstValue('id'))

def handle_init(conversation):
    conversation.addMediaTypeByName('text/plain')
    conversation.addMediaTypeByName('application/json')

def handle_get(conversation):
    id = get_id(conversation)

    session = get_session()
    try:
        note = session.query(Note).filter_by(id=id).one()
    except NoResultFound:
        return 404
    finally:
        session.close()

    conversation.modificationTimestamp = datetime_to_milliseconds(note.timestamp)
    return json.write(note.to_dict())

def handle_get_info(conversation):
    id = get_id(conversation)

    session = get_session()
    try:
        note = session.query(Note).filter_by(id=id).one()
    except NoResultFound:
        return None
    finally:
        session.close()

    return datetime_to_milliseconds(note.timestamp)

def handle_post(conversation):
    id = get_id(conversation)

    # Note: You can only "consume" the entity once, so if we want it
    # as text, and want to refer to it more than once, we should keep
    # a reference to that text.
    
    text = conversation.entity.text
    note_dict = json.read(text)

    session = get_session()
    try:
        note = session.query(Note).filter_by(id=id).one()
        note.update(note_dict)
        update_board_timestamp(session, note)
        session.flush()
    except NoResultFound:
        return 404
    finally:
        session.close()

    conversation.modificationTimestamp = datetime_to_milliseconds(note.timestamp)
    return json.write(note.to_dict())

def handle_delete(conversation):
    id = get_id(conversation)

    session = get_session()
    try:
        note = session.query(Note).filter_by(id=id).one()
        session.delete(note)
        update_board_timestamp(session, note, now())
        session.flush()
    except NoResultFound:
        return 404
    finally:
        session.close()

    return None

File: php/applications/stickstick/resources/data/note.php
<?php
require 'stickstick/data.php';

import java.lang.System;

function merge($key, $a, $b) {
	if(!in_array($key, $a)) {
		return $b[$key];
	}
	else {
		return $a[$key];
	}
}

function get_id($conversation) {
    return intval($conversation->locals['id']);

    //return intval($_GET['id']);
}

function handle_init($conversation) {
    $conversation->addMediaTypeByName('text/plain');
    $conversation->addMediaTypeByName('application/json');
}

function handle_get($conversation) {
	$id = get_id($conversation);

    $note = NULL;
    $connection = get_connection();
    print $connection;
    try {
        $note = get_note($id, $connection);
        if(is_null($note)) {
    		$connection->close();
        	return 404;
        }
    }
    catch(Exception $x) {
    	$connection->close();
    	throw $x;
    }
    $connection->close();

    $conversation->modificationTimestamp = $note['timestamp'];
    unset($note['timestamp']);
    return json_encode($note);
}

function handle_get_info($conversation) {
	$id = get_id($conversation);

    $note = NULL;
    $connection = get_connection();
    try {
        $note = get_note($id, $connection);
        if(is_null($note)) {
    		$connection->close();
        	return NULL;
        }
    }
    catch(Exception $x) {
    	$connection->close();
    	throw $x;
    }
    $connection->close();
   
    return $note['timestamp'];
}

function handle_post($conversation) {
	$id = get_id($conversation);

    // Note: You can only "consume" the entity once, so if we want it
    // as text, and want to refer to it more than once, we should keep
    // a reference to that text.
    
    $text = $conversation->entity->text;
    $note = json_decode($text, true);
    
    $connection = get_connection();
    try {
        $existing = get_note($id, $connection);
        if(is_null($existing)) {
    		$connection->close();
        	return 404;
        }
        $note = array_merge($existing, $note);
        update_note($note, $connection);
        update_board_timestamp($note, $connection);
    }
    catch(Exception $x) {
    	$connection->close();
    	throw $x;
    }
    $connection->close();

    $conversation->modificationTimestamp = $note['timestamp'];
    unset($note['timestamp']);
    return json_encode($note);
}

function handle_delete($conversation) {
	$id = get_id($conversation);

    $connection = get_connection();
    try {
        $note = get_note($id, $connection);
        if(is_null($note)) {
    		$connection->close();
        	return 404;
        }
        delete_note($note, $connection);
        update_board_timestamp($note, $connection, System::currentTimeMillis());
    }
    catch(Exception $x) {
    	$connection->close();
    	throw $x;
    }
    $connection->close();
    
    return NULL;
}
?>